zhusuan/distributions/univariate.py

```diff
@@ -163,6 +164,140 @@ def _log_prob(self, given):
+                                             (self._logstd, 'HalfNormal.logstd')])
+            std = tf.exp(self._logstd)
+            if check_numerics:
+                std = tf.check_numerics(std, "exp(logstd)")
+            self._std = std
+
+        try:
+            tf.broadcast_static_shape(self._mean.get_shape(),
+                                      self._std.get_shape())
+        except ValueError:
+            raise ValueError(
+                "mean and std/logstd should be broadcastable to match each "
+                "other. ({} vs. {})".format(
+                    self._mean.get_shape(), self._std.get_shape()))
+        self._check_numerics = check_numerics
+        super(HalfNormal, self).__init__(
+            dtype=dtype,
+            param_dtype=dtype,
+            is_continuous=True,
+            is_reparameterized=is_reparameterized,
+            group_event_ndims=group_event_ndims)
+
+    @property
+    def mean(self):
+        """The mean of the Normal distribution."""
+        return self._mean
+
+    @property
+    def logstd(self):
+        """The log standard deviation of the Normal distribution."""
+        return self._logstd
+
+    @property
+    def std(self):
+        """The standard deviation of the Normal distribution."""
+        return self._std
+
+    def _value_shape(self):
+        return tf.constant([], dtype=tf.int32)
+
+    def _get_value_shape(self):
+        return tf.TensorShape([])
+
+    def _batch_shape(self):
+        return tf.broadcast_dynamic_shape(tf.shape(self.mean),
+                                          tf.shape(self.std))
+
+    def _get_batch_shape(self):
+        return tf.broadcast_static_shape(self.mean.get_shape(),
+                                         self.std.get_shape())
+
+    def _sample(self, n_samples):
+        mean, std = self.mean, self.std
+        if not self.is_reparameterized:
+            mean = tf.stop_gradient(mean)
+            std = tf.stop_gradient(std)
+        shape = tf.concat([[n_samples], self.batch_shape], 0)
+        samples = tf.abs(tf.random_normal(shape, dtype=self.dtype) * std + mean)
+        static_n_samples = n_samples if isinstance(n_samples, int) else None
+        samples.set_shape(
+            tf.TensorShape([static_n_samples]).concatenate(
+                self.get_batch_shape()))
+        return samples
+
+    def _log_prob(self, given):
+        #TODO: Raise Error
+        c = -0.5 * np.log(2 * np.pi)
+        precision = tf.exp(-1 * self.logstd)
+        if self._check_numerics:
+            precision = tf.check_numerics(precision, "precision")
+        temp = given >= 0
+        temp = tf.cast(temp, tf.float32)
+        #return c - self.logstd - 0.5 * precision * tf.square(given - self.mean)
+        return (0.5 * (tf.log(2.0) - tf.log(np.pi)) - self.logstd - 0.5 * tf.square(given * tf.exp(-1 * self.logstd))) + tf.log(temp)
+    def _prob(self, given):
+        return tf.exp(self._log_prob(given))
+
 
```

Comment:
A test should be added in tests/distributions/test_univariate.py. Following other distributions.